substrate/frame/staking/src/mock.rs

```diff
@@ -598,6 +603,18 @@ pub(crate) fn bond_nominator(who: AccountId, val: Balance, target: Vec<AccountId
 	assert_ok!(Staking::nominate(RuntimeOrigin::signed(who), target));
 }
 
+pub(crate) fn bond_virtual_nominator(
+	who: AccountId,
+	payee: AccountId,
+	val: Balance,
+	target: Vec<AccountId>,
+) {
+	// who is provided by another pallet in a real scenario.
+	System::inc_providers(&who);
```

Comment:
A bit fishy. Can you explain more? if `who` has enough ED, that should be a provider?

Reply:
who is a keyless acc in a real scenario and another pallet (delegation staking pallet) provides for it. Improved the comment.